model/change_point_detection.go

```diff
@@ -462,6 +464,65 @@ func createChangePoint(ctx context.Context, env cedar.Environment, resultToUpdat
 	return errors.Wrap(err, "Unable to create change point")
 }
 
+type ChangePointInfo struct {
+	PerfResultID string `json:"perf_result_id"`
+	Measurement  string `json:"measurement"`
+}
+
+func TriageChangePoints(ctx context.Context, env cedar.Environment, changePoints []ChangePointInfo, status TriageStatus) error {
+	coll := env.GetDB().Collection(perfResultCollection)
+
+	var conditions []bson.M
+	for _, stub := range changePoints {
+		conditions = append(conditions, bson.M{
+			perfIDKey: stub.PerfResultID,
+			bsonutil.GetDottedKeyName(perfAnalysisKey, perfAnalysisChangePointsKey, perfChangePointMeasurementKey): stub.Measurement,
+		})
+	}
+	filter := bson.M{
+		"$or": conditions,
+	}
+
+	cur, err := coll.Find(ctx, filter)
+	if err != nil {
+		return errors.Wrap(err, "Could not execute query finding change points for triage")
+	}
+	var results []PerformanceResult
+	if err := cur.All(ctx, &results); err != nil {
+		return errors.Wrap(err, "Could not decode performance results for triage")
+	}
+
+	for _, stub := range changePoints {
+		for _, res := range results {
+			if res.ID == stub.PerfResultID {
+				for _, cp := range res.Analysis.ChangePoints {
+					if cp.Measurement == stub.Measurement {
+						goto cont
```

Comment:
avoid using gotos since they make it really hard to tell where in the execution stack you are. Instead add a loop label to the change points loop and continue to that here

Reply:
Neat, I didn't know you could do that in Go!